libs/main/connection-window/src/connection-window.component.ts

```diff
@@ -52,7 +54,11 @@ export class ConnectionWindowComponent extends SubscriptionHandler implements On
 
     if (this.configs?.length > 0) {
       // get last saved config
-      this.form.setValue(this.configs[this.configs.length - 1]);
+      const lastConfig = this.configs[this.configs.length - 1];
+      this.form.setValue(lastConfig);
+
+      // Restore SSL preference if it was saved with the config
+      this.sslEnabled = !!lastConfig.sslEnabled;
```

Comment:
it's not critical but why not just include `sslEnabled` inside the form instead of keeping it outside of it?